Added language filter for the category meta box

chandonn-zz opened the pull request:
Fixes #38

chandonn committed:
Added language filter for the category meta box

chandonn committed:
Added documentation for the last commited function

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
+	*/
+	public function monk_apply_terms_filter( $term_query ) {
+		$screen = get_current_screen();
+		if ( 'edit' === $screen->parent_base ) {
```
Add conditional to check if is category query
Add conditional to check `$lang` query var

chandonn-zz replied:
Fixed

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
+		if ( 'edit' === $screen->parent_base ) {
+			$post_id   = get_the_id();
+			$meta_lang = sanitize_title( get_post_meta( $post_id, '_monk_post_language', true ) );
+			$term_args = array(
```
Remove default args

chandonn-zz replied:
Fixed

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
+				'hide_empty' => false,
+				'fields'     => 'all',
+				'count'      => true,
+				'meta_key'   => 'monk-language',
```
Use `_monk_term_language` meta key

chandonn-zz replied:
Fixed

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -231,7 +231,7 @@ public function monk_post_meta_box_field_render( $post ) {
 		wp_nonce_field( basename( __FILE__ ), 'monk_post_meta_box_nonce' );
 
-		if ( $post_default_language == '' ) {
+		if ( '' == $post_default_language ) {
```
Use `===`

chandonn-zz replied:
Fixed

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
+	 *
+	 * @since    1.0.0
+	*/
+	public function monk_apply_terms_filter( $term_query ) {
```
Modify function name to `monk_category_language_filter`

chandonn-zz replied:
Fixed

brenoalvs reviewed (changes requested):
@chandonn take a look

chandonn committed:
Fixed code standards issue

Changed function name

Added conditional to check taxonomy type

Changed meta_key name usage

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -298,6 +298,37 @@ public function monk_save_post_meta_box( $post_id ) {
+			$meta_lang  = sanitize_title( get_post_meta( $post_id, '_monk_post_language', true ) );
+			$query_lang = $_GET['lang'];
+			
+			if ( isset( $meta_lang ) && '' !== $meta_lang ) {
```
Use `||`

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -298,6 +298,37 @@ public function monk_save_post_meta_box( $post_id ) {
+			$term_args  = '';
+			$post_id    = get_the_id();
+			$meta_lang  = sanitize_title( get_post_meta( $post_id, '_monk_post_language', true ) );
+			$query_lang = $_GET['lang'];
```
Add sanitize function

brenoalvs reviewed (changes requested):
@chandonn

chandonn committed:
Changed category filter logic

chandonn committed:
Added sanitizer on lang query var
